symbolic/src/Data/Macaw/Symbolic/Memory.hs

```diff
@@ -321,23 +365,24 @@ populateMemory :: ( CB.IsSymBackend sym bak
                -- ^ The symbolic backend
                -> MemoryModelContents
                -- ^ A flag to indicate how to populate the memory model based on the memory image
-               -> MC.Memory (MC.ArchAddrWidth arch)
+               -> [MC.Memory (MC.ArchAddrWidth arch)]
```

Comment:
I suggest making this a `Foldable` container (so that we can pass in the `NonEmpty` list from the new entry point easily)